internal/querycoord/task.go

```diff
@@ -1935,8 +1935,8 @@ func (lbt *loadBalanceTask) processNodeDownLoadBalance(ctx context.Context) erro
 			log.Info("loadBalanceTask: get collection's all partitionIDs", zap.Int64("collectionID", collectionID), zap.Int64s("partitionIDs", toRecoverPartitionIDs))
 			replica, err := lbt.getReplica(nodeID, collectionID)
 			if err != nil {
-				lbt.setResultInfo(err)
-				return err
+				// getReplica maybe failed, it will cause the balanceTask execute infinitely
+				log.Warn("loadBalanceTask: get replica failed", zap.Int64("collectionID", collectionID), zap.Int64("nodeId", nodeID))
```

Comment:
Why `getReplica` failed at the first place?

Reply:
If `getReplica` returns error, it's not safe to use `replica` any more(it shall be nil).

Reply:
> Why `getReplica` failed at the first place?

Inconsistency happens if the QueryCoord crashes while the LoadBalance task has modified the replica, but itself is not removed from etcd. The recovered task will never get replica with the nodeID and collectionID, and retry infinitely.